### ISSUE TITLE
Unclear documentation for WDATP custom detections

### ISSUE THREAD

**Issue by KevinBollengier** [MicrosoftDocs/windows-itpro-docs#4904]
There should more clear documentation related to the creation of custom detections with Windows Defender ATP.. I keep getting error messages that tell me certain columns such as EventTime, MachineID and ReportID are lacking.. 

Using the project statement does not resolve this issue either...

---
#### Documentdetails

⚠ *Dit gedeelte niet bewerken. Het is vereist om problemen te koppelen tussen docs.microsoft.com en GitHub.*

* ID: 93d8d7cc-6b88-b07f-ad25-6439cc0292b1
* Version Independent ID: f2d25adc-f92e-dbac-4c2c-d678684f0de3
* Content: [Create custom detection rules in Microsoft Defender ATP](https://docs.microsoft.com/nl-nl/windows/security/threat-protection/microsoft-defender-atp/custom-detection-rules)
* Content Source: [windows/security/threat-protection/microsoft-defender-atp/custom-detection-rules.md](https://github.com/MicrosoftDocs/windows-itpro-docs/blob/master/windows/security/threat-protection/microsoft-defender-atp/custom-detection-rules.md)
* Product: **w10**
* Technology: **windows**
* GitHub Login: @mjcaparas
* Microsoft Alias: **macapara**

**Comment by KevinBollengier** [comment]
As extra the working query I can't seem to get into a custom alert:

let remotePortCountThreshold = 10; 
NetworkCommunicationEvents
| where EventTime > ago(7d) and RemoteIP startswith "172.16" or RemoteIP startswith "192.168" or RemoteIP startswith "10." 
| where InitiatingProcessFileName contains "nmap" or InitiatingProcessFileName contains "scan" or InitiatingProcessFileName contains "zenmap" or InitiatingProcessFileName contains "ncat" 
| summarize
    by ComputerName, RemoteIP, RemotePort, InitiatingProcessFileName
| summarize RemotePortCount=dcount(RemotePort) by ComputerName, RemoteIP, InitiatingProcessFileName
| where RemotePortCount > remotePortCountThreshold

**Comment by mypil** [comment]
@KevinBollengier - Thank you for submitting feedback.

I will get this issue over to the Win10 ITPro writing team for investigation.

Thank you for reporting and making the docs better. Much appreciated.

I made a note to request the team to update this when the work is complete.

**Comment by iadgovuser1** [comment]
The result set needs to return the EventTime, MachineID and ReportID  columns as part of the data. You aren't returning those due to your summarize statement. It isn't enough to project those columns in this case because summarize doesn't return what you project. It returns what you summarize.

**Comment by KevinBollengier** [comment]
> The result set needs to return the EventTime, MachineID and ReportID columns as part of the data. You aren't returning those due to your summarize statement. It isn't enough to project those columns in this case because summarize doesn't return what you project. It returns what you summarize.

I've tried this too, but then I got syntax errors.. (In my defense, I just got started with the WDATP, but it's hard to find good documentation...)

**Comment by joinimran** [comment]
@KevinBollengier Thank you for sharing your feedback. Can you please let us know what part needs modification/updation so that we can update the document and make it more useful. 
Imran.

**Comment by KevinBollengier** [comment]
@joinimran Hi,in the meantime I've found my way a bit more due to trial and error. But for me it was not clear why I couldn't save certain queries as custom detections. I've included one of such queries above as I wanted to detect the use of certain network mappers. The query works, but somehow it can't be saved as a custom detection.

Something along the lines that certain columns are not returned?

**Comment by joinimran** [comment]
Thank you for updating this. Did you get a chance to open a support case with Microsoft so that technical support teams can look into this and see why your query was not working? If you haven't opened up a case, please do so, this will help the product development team to investigate the issue and if there will be any requirement to update the document, that will be taken care from the product team or you can share the resolutions so that we can also update the document accordingly.  

Imran.

**Comment by iadgovuser1** [comment]
@KevinBollengier  If the columns rendered from running the query do not include EventTime, MachineID and ReportID columns at a minimum, then you will not be able to create a custom detection based on the query. That is the issue here. This requirement is documented at https://docs.microsoft.com/en-us/windows/security/threat-protection/microsoft-defender-atp/custom-detection-rules#create-a-custom-detection-rule

**Comment by KevinBollengier** [comment]
@iadgovuser1, Hi there, just wondering in that case how I would fix it with the aggregate colums (count/dcount)...

**Comment by iadgovuser1** [comment]
@KevinBollengier I don't believe it is possible to create a custom detection for queries that use summarize because custom detections work on a specific event instances and summarize queries are summaries of a bunch of events. This may be as close as you can get:
```
NetworkCommunicationEvents
| project EventTime, MachineId, ReportId, ComputerName, RemoteIP, InitiatingProcessFileName
| where EventTime > ago(7d) and RemoteIP startswith "172.16" or RemoteIP startswith "192.168" or RemoteIP startswith "10."
| where InitiatingProcessFileName contains "nmap" or InitiatingProcessFileName contains "scan" or InitiatingProcessFileName contains "zenmap" or InitiatingProcessFileName contains "ncat" 
```

**Comment by lomayor** [comment]
@KevinBollengier, I got this from one of our devs. The last line get only the latest the EventTime and ReportId of the latest event among multiple events returned by the query and adds the count by MachineId.

`MiscEvents
| where EventTime > ago(7d)
| where ActionType == "AntivirusDetection"
| summarize (EventTime, ReportId)=arg_max(EventTime, ReportId), count() by MachineId`

I'll add this to the doc.

**Comment by mypil** [comment]
@KevinBollengier - From our understanding, the issue has been resolved based on the merged commit [[8c271fc](https://github.com/MicrosoftDocs/windows-itpro-docs/commit/8c271fc3375a8b9481b2f3e96a5fb7cf94ad795c)] but it may take a few days for the merged content to appear in the article. If you feel it hasn't been resolved, please re-open this issue.

Thank you for your contribution to make the docs better! Much appreciated!
